storage/local/crashrecovery.go

```diff
@@ -42,6 +42,12 @@ func (p *persistence) recoverFromCrash(fingerprintToSeries map[clientmodel.Finge
 	count := 0
 	seriesDirNameFmt := fmt.Sprintf("%%0%dx", seriesDirNameLen)
 
+	// Delete the fingerprint mapping file as it might be stale or
+	// corrupt. We'll rebuild the mappings as we go.
```

Comment:
The stochastic assignment happens when the samples come in for the first time and are mapped or not.

Here we are not dealing with sample ingestion. The mapping has already happening. We are just re-constructing it from the indices. (In other words: The indices always use _mapped_ fingerprints.)